File: 백준/Silver/14402. 야근/야근.py
import sys
from collections import defaultdict
input = sys.stdin.readline

rooms = defaultdict(int)
count = 0
for _ in range(int(input())):
    name, pm = input().split()

    if pm == "+":
        rooms[name] += 1
    else:
        if rooms[name]:
            rooms[name] -= 1
        else:
            count += 1

count += sum(rooms.values())
print(count)

File: 백준/Silver/14402. 야근/야근.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int q = Integer.parseInt(br.readLine());

        Map<String, Integer> rooms = new HashMap<>();
        long count = 0;

        for (int i = 0; i < q; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String name = st.nextToken();
            String pm = st.nextToken();

            rooms.putIfAbsent(name, 0);

            if (pm.equals("+"))
                rooms.put(name, rooms.get(name) + 1);
            else if (rooms.get(name) > 0)
                rooms.put(name, rooms.get(name) - 1);
            else
                count++;
        }

        for (int v : rooms.values())
            count += v;

        System.out.println(count);
    }
}